server/server.go

```diff
@@ -544,6 +546,17 @@ func (s *Server) handlePublishWithoutResponse(r *http.Request, v *visitor) (*mes
 	if err != nil {
 		return nil, err
 	}
+
+	var v_billing *visitor
+	if strings.HasPrefix(t.ID, unifiedpushTopicPrefix) {
```

Comment:
Here's an idea: instead of using the prefix, could we use the presence of the `?up=1` query param? If the qurey param is present, bill differently? I vaguely remember this being discussed before, but I do not recall what the outcome was

Reply:
From the UP dev chat

> up=1 is sent by the server, whereas these messages are being billed to the subscriber. I thought about that too, but then came up with this:
> 
>   -  I am subscribed to [ntfy.sh/videogameupdates](http://ntfy.sh/videogameupdates), an open channel for anyone to post
>   -  A malicious actor comes along and sends 10000 messages to [ntfy.sh/videogameupdates?up=1](http://ntfy.sh/videogameupdates?up=1)
>   - Me, poor subscriber, is now rate-limited 
> 
> So, we need a way in which the _subscriber knows that they are subscribing to a subscriber-billed topic_ instead of trusting the sender to inform us. The easiest way is the topic.